hexrd/instrument.py

```diff
@@ -165,6 +171,77 @@ def _fix_indices(idx, lo, hi):
     return nidx
 
 
+def _parse_imgser_dict(imgser_dict, det_key, roi=None):
+    """
+    Associates a dict of imageseries to the target panel(s).
+
+    Parameters
+    ----------
+    imgser_dict : dict
+        The input dict of imageseries.  Either `det_key` is in imgser_dict, or
+        the shared key is.  Entries can be an ImageSeries object or a 2- or 3-d
+        ndarray of images.
+    det_key : str
+        The target detector key.
+    roi : tuple or None, optional
+        The roi of the target images.  Format is
+            ((row_start, row_stop), (col_start, col_stop))
+        The stops are used in the normal sense of a slice. The default is None.
+
+    Raises
+    ------
+    RuntimeError
+        If niether `det_key` nor the shared key is in the input imgser_dict;
+        Also, if the shared key is specified but the roi is None.
+
+    Returns
+    -------
+    ims : hexrd.imageseries
+        The desired imageseries object.
+
+    """
+    # grab imageseries for this detector
+    try:
+        ims = imgser_dict[det_key]
+    except(KeyError):
+        if multi_ims_key in imgser_dict:
+            if roi is None:
+                raise RuntimeError(
+                    "roi must be specified to use shared imageseries"
+                )
+            images_in = imgser_dict[multi_ims_key]
+            if isinstance(images_in, ims_classes):
```

Comment:
```suggestion
            if isinstance(images_in, ImageSeries):
```

I believe the above will work because `ProcessedImageSeries` and `OmegaImageSeries` both subclass `ImageSeries`, which means that they will return `True` for `isinstance(images_in, ImageSeries)`. So I think we can remove the `ims_classes` variable.

Reply:
You are right; I'll make the edit.